test/utils_test.js

```diff
@@ -168,7 +168,7 @@ describe("utils", () => {
         expect(
           getDefaultFormState(schema, { level1: { level2: { leaf4: 4 } } })
         ).eql({
-          level1: { level2: { leaf1: 1, leaf2: 2, leaf3: 3, leaf4: 4 } },
+          level1: { level2: { leaf1: 1, leaf2: 1, leaf3: 1, leaf4: 4 } },
```

Comment:
See, this doesn't make any sense to me. I would have expected `{ level1: { level2: { leaf3: 3, leaf4: 4 } } }`. I would have expected `level1.default` not to apply because there is already a `level1` field, and `level1.level2.default` to not apply because there is already a `level1.level2` field, and `level1.level2.leaf3.default` to apply because there is no `level1.level2.leaf3`. Could you summarize under what circumstances a default is filled in?

Reply:
I agree @glasserc